### PR TITLE
archive: fix unrar-free and add support for unar

### PR DESCRIPTION
**Pull request by corna**

### CODE REVIEW

File: modules/archive/functions/lsarchive [hunk 1]
```diff
@@ -43,9 +43,11 @@ while (( $# > 0 )); do
       || lzcat "$1" | tar x${verbose:+v}f - ;;
     (*.tar) tar t${verbose:+v}f "$1" ;;
     (*.zip|*.jar) unzip -l${verbose:+v} "$1" ;;
-    (*.rar) unrar &> /dev/null \
+    (*.rar) unrar -V &> /dev/null \
```
Comment (belak) [inline modules/archive/functions/lsarchive]: It would probably be better to use a check like `(( ${+commands[unrar]} ))` here and for the other commands.

**Comment by corna** [comment]
Done

**Comment by belak** [comment]
I'll try to take a look at this again soon. Thanks!

The only thing I saw at a glance is that the first option for unrar isn't surrounded by `()` so it's not as easy to follow as I'd like, but it's not a deal breaker.

**Comment by belak** [comment]
Could you add a note that the `archive` function won't work for rar files if you only have `lsar` and `unar` (and don't have any of the other alternatives).

Are there any other filetypes this would work with, since as far as I can tell these commands are from the Unarchiver (maybe not something for this PR, but it may be worth exploring in the future).

**Comment by corna** [comment]
Done.

Yes, The Unarchiver can be also used for other files.

**Comment by belak** [comment]
Looks good! Thanks!

**Pull request merged**